app/models/os_major_version.rb

```diff
@@ -4,11 +4,17 @@
 class OsMajorVersion < ApplicationRecord
   OS_MAJOR_VERSION = Arel.sql(
     <<-SQL.gsub("\n", ' ').squeeze(' ')
-      REPLACE(ref_id, '#{Xccdf::Benchmark::REF_PREFIX}-', '')::int
-      AS os_major_version
+      REPLACE("benchmarks"."ref_id", '#{Xccdf::Benchmark::REF_PREFIX}-', '')::int
+      AS "os_major_version"
     SQL
   )
 
+  BM_DESC = "#{Xccdf::Benchmark::SORT_BY_VERSION} DESC"
+  PROFILE_COLS = Profile.column_names.reject { |c| c == 'ref_id' }.map do |column|
+    "(array_agg(\"profiles\".\"#{column}\" ORDER BY #{BM_DESC}))[1] as \"#{column}\""
```

Comment:
You cannot really do this, anything non-aggregated in SELECT must appear in GROUP BY :confused: also we need the ID of the latest supported profile. But I can surely convert things to arel.

Reply:
Oh sorry, I keep forgetting that this was possible only within MySQL...
I did some quick research, and found other option that work with Postgres. The answer might be the use of JOIN to retrieve and append aggregated data.

Observe this SQL:
```
SELECT p.*, grp.bm_versions
FROM "profiles" AS p
INNER JOIN "benchmarks" AS b ON b."id" = p."benchmark_id"
INNER JOIN (
    SELECT "profiles".ref_id AS profile_ref_id, benchmarks.ref_id AS benchmark_ref_id, array_agg(benchmarks.version) AS bm_versions
    FROM "profiles" 
    INNER JOIN "benchmarks" ON "benchmarks"."id" = "profiles"."benchmark_id" 
    WHERE "profiles"."parent_profile_id" IS NULL 
    AND "benchmarks"."ref_id" = 'xccdf_org.ssgproject.content_benchmark_RHEL-7'
    GROUP BY profile_ref_id, benchmark_ref_id
) AS grp ON grp.profile_ref_id = p.ref_id AND grp.benchmark_ref_id = b.ref_id
WHERE p."parent_profile_id" IS NULL 
AND b."ref_id" = 'xccdf_org.ssgproject.content_benchmark_RHEL-7';
```
If we extract the inner-join subselect it might be more readable.

WDYT?

Reply:
Isn't this way more complicated than the current one? :thinking:

Reply:
What wanted to avoid is:
* the enumeration of all profile fields
* need to have these fields in aggregation
* and having each aggregated field to order.

My latest proposal seems a bit more logical. First it retrieves records, and then it appends extra and aggregated data to each corresponding row.

Reply:
```
grp = Profile.canonical.joins(:benchmark).select(:ref_id, 'benchmarks.ref_id AS benchmark_ref_id', 'array_agg(benchmarks.version) AS bm_versions').group(:ref_id, 'benchmarks.ref_id')
OsMajorVersion.first.profiles.joins("INNER JOIN (#{grp.to_sql}) AS grp ON grp.ref_id = profiles.ref_id AND benchmarks.ref_id = grp.benchmark_ref_id").select('profiles.*, grp.bm_versions')
```